[Rayner Lim Fang Yuh] iP

rayray39 opened the pull request:
# Janet 💁🏻‍♀️ 
> Janet is the informational assistant in The Good Place, played by D'Arcy Carden. [Wiki link](https://en.wikipedia.org/wiki/D%27Arcy_Carden)

Janet frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~miserable~~ ***DELIGHTFUL*** to use

All you need to do is,
1. download it from [here](https://github.com/rayray39/ip/releases/download/A-Jar/janet.jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you

And it is **FREE**!

Features
- [x] Managing tasks
- [x] Mark/Unmark tasks
- [x] Delete tasks
- [x] Find specific tasks based on keyword
- [ ] GUI (coming soon!)

#

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 

```java
public class Main {
    public static void main(String[] args) {
        new Janet("janet.txt").run();
    }
}
```

raynerlimfangyuh committed:
Main.java: Add method to load .txt content into ArrayList

raynerlimfangyuh committed:
Main.java: Add deadline and event to textFileToArrayList

raynerlimfangyuh committed:
Main.java: Remove commented out code

raynerlimfangyuh committed:
Main.java: Add javadoc to textFileToArrayList

raynerlimfangyuh committed:
Main.java: Update findDeadlineDetails for LocalDate

raynerlimfangyuh committed:
Main.java: Remove commented code

raynerlimfangyuh committed:
Main.java: Update findEventDetails for LocalDate

raynerlimfangyuh committed:
Main.java: Remove commented code

raynerlimfangyuh committed:
Main.java: Update findDeadlineDetails to take in time

raynerlimfangyuh committed:
Main.java: Update javadoc of findDeadlineDetails

raynerlimfangyuh committed:
Main.java: Update findEventDetails to take in time

raynerlimfangyuh committed:
Deadline.java: Update datatype of dueDate to LocalDateTime

raynerlimfangyuh committed:
Deadline.java: Update whitespace separator to - in dueDate

raynerlimfangyuh committed:
Main.java: Add creation of Deadline with LocalDateTime

raynerlimfangyuh committed:
Deadline.java: Update delimeter of dueDate to whitespace

raynerlimfangyuh committed:
Main.java: Add method createDeadlineFromJanetTextFile

raynerlimfangyuh committed:
Deadline.java: Update toString of Deadline

raynerlimfangyuh committed:
Main.java: Update DEADLINE case to accept LocalDateTime

raynerlimfangyuh committed:
Main.java: Remove commented code in textFileToArrayList

raynerlimfangyuh committed:
Main.java: Add javadoc to createDeadlineFromJanetTextFile

raynerlimfangyuh committed:
Event.java: Update startDate and endDate to LocalDateTime type

raynerlimfangyuh committed:
Main.java: Remove unused import package

raynerlimfangyuh committed:
Main.java: Add method createEventFromJanetTextFile

raynerlimfangyuh committed:
Main.java: Add javadoc to createEventFromJanetTextFile

raynerlimfangyuh committed:
Main.java: Update findEventDetails to convert user input to datetime format

raynerlimfangyuh committed:
Main.java: Update switch case for EVENT to accept LocalDateTime

raynerlimfangyuh committed:
Main.java: Update outputEndTime to outputEndDate in findEventDetails

raynerlimfangyuh committed:
Main.java: Add description to javadoc of all public methods

raynerlimfangyuh committed:
Main.java: Add method createToDoFromJanetTextFile

raynerlimfangyuh committed:
Main.java: Add javadoc to method createToDoFromJanetTextFile

raynerlimfangyuh committed:
Main.java: Update textFileToArrayList to use createToDoFromJanetTextFile

raynerlimfangyuh committed:
Storage.java: Add methods for loading and reading of .txt file

raynerlimfangyuh committed:
Main.java: Remove methods for loading and reading of .txt file

raynerlimfangyuh committed:
Storage.java: Update switch case in textFileToArrayList to follow code style

raynerlimfangyuh committed:
Main.java: Update comments in main method

raynerlimfangyuh committed:
TaskList.java: Add methods to operate on listOfTasks

raynerlimfangyuh committed:
TaskList.java: Add getter for task in listOfTasks

raynerlimfangyuh committed:
Janet.java: Update methods to use TaskList

raynerlimfangyuh committed:
Janet.java: Update deleteTask to remove correct task from list

raynerlimfangyuh committed:
Parser.java: Add methods to parse lines for deadline and event

raynerlimfangyuh committed:
Main.java: Remove findDeadlineDetails and findEventDetails

raynerlimfangyuh committed:
TaskList.java: Add boolean method isEmpty for listOfTasks

raynerlimfangyuh committed:
Storage.java: Update catch clause to print message instead

raynerlimfangyuh committed:
Ui.java: Add methods to print different command messages

raynerlimfangyuh committed:
Main.java: Update main method to use Ui and TaskList instead of Janet

raynerlimfangyuh committed:
Main.java: Remove unused import packages

raynerlimfangyuh committed:
Janet.java: Remove methods that are a duplicate of TaskList's

raynerlimfangyuh committed:
Main.java: Remove commented out code

raynerlimfangyuh committed:
Storage.java: Update textFileToArrayList to throw JanetException

raynerlimfangyuh committed:
Main.java: Update main method to surround taskList with try-catch

raynerlimfangyuh committed:
Ui.java: Add showLoadingError method

raynerlimfangyuh committed:
Janet.java: Remove duplicated code from Ui.java and TaskList.java

raynerlimfangyuh committed:
Deadline.java: Update constructor to use static methods

raynerlimfangyuh committed:
Deadline.java: Add comments into constructors

raynerlimfangyuh committed:
Deadline.java: Update field to accept .dueDate instead of getter

raynerlimfangyuh committed:
Deadline.java: Remove unused getter, getDueDateAndTime

raynerlimfangyuh committed:
Event.java: Update constructor to use static methods

raynerlimfangyuh committed:
Event.java: Add comment to constructor

raynerlimfangyuh committed:
Parser.java: Remove duplicate methods and unused package imports

raynerlimfangyuh committed:
Main.java: Update event and deadline to be created by new constructors

raynerlimfangyuh committed:
ToDo.java: Add static createToDo method and new constructor

raynerlimfangyuh committed:
ToDo.java: Add comments into constructors

raynerlimfangyuh committed:
Main.java: Update todo to be created by new constructors

raynerlimfangyuh committed:
Main.java: Add comment into switch case for todo

raynerlimfangyuh committed:
CommandType.java: Add BYE and LIST as new commands in enum

raynerlimfangyuh committed:
Parser.java: Add methods for exception handling and processing user inputs

raynerlimfangyuh committed:
Main.java: Remove duplicate methods for exception handling

raynerlimfangyuh committed:
Janet.java: Update run method using code from Main.main()

raynerlimfangyuh committed:
Janet.java: Add boolean flag hasTypedBye to break out of while loop

raynerlimfangyuh committed:
Delete Main.java and Command.java

raynerlimfangyuh committed:
Merge branch 'branch-A-MoreOOP'

raynerlimfangyuh committed:
Parser.java: Add new if-clause to handle empty input lines

raynerlimfangyuh committed:
Janet.java: Move Janet.java into janet package

raynerlimfangyuh committed:
Move all files into janet package

raynerlimfangyuh committed:
Merge branch 'branch-A-Packages'

raynerlimfangyuh committed:
Merge branch 'add-gradle-support'

raynerlimfangyuh committed:
build.gradle: Update parameter in mainClass.set() to janet.Janet

raynerlimfangyuh committed:
branch-A-Gradle

raynerlimfangyuh committed:
Merge branch 'branch-A-Gradle'

raynerlimfangyuh committed:
Storage.java: Remove commented code

raynerlimfangyuh committed:
Storage.java: Add comments

raynerlimfangyuh committed:
Storage.java: Update saveToJanetTextFile to throw JanetException

raynerlimfangyuh committed:
Ui.java: Add showSavingError to print message when error in saving .txt file

raynerlimfangyuh committed:
Janet.java: Add try-catch to storage.saveToJanetTextFile in run method

raynerlimfangyuh committed:
Parser.java: Add comments for clarity and remove unneeded comments

raynerlimfangyuh committed:
Parser.java: Update if clause in checkInaccurateCommand to be more general

raynerlimfangyuh committed:
Parser.java: Add method getCommandTypes

raynerlimfangyuh committed:
Parser.java: Update if-clause in checkInaccurateCommand to be more general

raynerlimfangyuh committed:
Janet.java: Remove unneeded throw IOException statement

raynerlimfangyuh committed:
Parser.java: Add methods to validate user command for deadline & event

raynerlimfangyuh committed:
Parser.java: Add checkUserInput method to run all checks together

raynerlimfangyuh committed:
Parser.java: Add validateByeAndList method and add into checkUserInput

raynerlimfangyuh committed:
Janet.java: Update try block to use Parser.checkUserInput and remove comments

raynerlimfangyuh committed:
Add DeadlineTest and EventTest to JUnit testing

raynerlimfangyuh committed:
Delete DeadlineTest and replace with ParserTest

raynerlimfangyuh committed:
EventTest.java: Add method to test successful instantiation of Event object

raynerlimfangyuh committed:
EventTest.java: Update test to give correct output

raynerlimfangyuh committed:
ParserTest.java: Add test methods to check correct exception messages

raynerlimfangyuh committed:
Merge branch 'branch-A-JUnit'

raynerlimfangyuh committed:
Event.java: Add JavaDoc to class and public method createEventCommand

raynerlimfangyuh committed:
Parser.java: Add JavaDoc to all public methods

raynerlimfangyuh committed:
ToDo.java: Add JavaDoc to public methods and class

raynerlimfangyuh committed:
Deadline.java: Add JavaDoc to public methods and class

raynerlimfangyuh committed:
Task.java: Add JavaDoc to class

raynerlimfangyuh committed:
TaskList.java: Add JavaDoc to public methods and class

raynerlimfangyuh committed:
TaskList.java: Remove unneeded comment

raynerlimfangyuh committed:
Ui.java: Add JavaDoc to public methods and class

raynerlimfangyuh committed:
Parser.java: Remove unused import statement

raynerlimfangyuh committed:
Parser.java: Add JavaDoc to class

raynerlimfangyuh committed:
Storage.java: Add JavaDoc to class

raynerlimfangyuh committed:
CommandType.java: Add FIND value to enum

raynerlimfangyuh committed:
Ui.java: Add showFindMessage method to show matching tasks

raynerlimfangyuh committed:
Ui.java: Add displayTasksInList method to show list of tasks

raynerlimfangyuh committed:
Ui.java: Update showTasks to use correct TaskList getter

raynerlimfangyuh committed:
Ui.java: Update if-clause in displayTasksInList method

raynerlimfangyuh committed:
Ui.java: Update displayTasksInList to be class method

raynerlimfangyuh committed:
Janet.java: Add case for FIND in run method

raynerlimfangyuh committed:
TaskList.java: Add findTask method to find tasks that contains keyword

raynerlimfangyuh committed:
Janet.java: Add logic to find task case and add comments

raynerlimfangyuh committed:
Ui.java: Add message when taskList in showFindMessage is empty

raynerlimfangyuh committed:
Parser.java: Update logic in fromOrToKeywordNotFound

raynerlimfangyuh committed:
Ui.java: Update code to surround message in else-clause

raynerlimfangyuh committed:
Merge branch 'branch-A-JavaDoc'

raynerlimfangyuh committed:
Merge branch 'branch-Level-9'

raynerlimfangyuh committed:
Ui.java: Resolve merge conflicts when merging to master

raynerlimfangyuh committed:
Ui.java: Add JavaDoc to public methods

raynerlimfangyuh committed:
Ui.java: Change constant name to screaming snake case

raynerlimfangyuh committed:
Ui.java: Change long print statements to break after operator

raynerlimfangyuh committed:
Ui.java: Resolve merge conflicts between branch-A-CodingStandard and master

raynerlimfangyuh committed:
Ui.java: Update constant horizontalLine to HORIZONTAL_LINE

raynerlimfangyuh committed:
TaskList.java: Add JavaDoc to findTask method

raynerlimfangyuh committed:
TaskList.java: Update findTask to findTasks for coding standard

raynerlimfangyuh committed:
Janet.java: Update usage of findTask to findTasks

raynerlimfangyuh committed:
CommandType.java: Add JavaDoc to class

raynerlimfangyuh committed:
JanetException.java: Add JavaDoc to class

raynerlimfangyuh committed:
Janet.java: Add JavaDoc to class and public methods

raynerlimfangyuh committed:
Janet.java: Remove unneeded throw IOException in main method

raynerlimfangyuh committed:
Add exception handling for parsing LocalDate and LocalTime

raynerlimfangyuh committed:
Add custom DateAndTimeFormatter to eliminate repeated code

raynerlimfangyuh committed:
Shift DateAndTimeFormatter method from Event into Task

raynerlimfangyuh committed:
Deadline.java: Update findDeadlineDetails to use DateAndTimeFormatter

raynerlimfangyuh committed:
Event.java: Update findEventDetails to explicitly call Task.DateAndTimeFormatter

raynerlimfangyuh committed:
Event.java: Update findEventDetails to catch ArrayIndexOutOfBoundsException too

raynerlimfangyuh committed:
Parser.java: Remove unneeded validateEvent method and logic

raynerlimfangyuh committed:
Parser.java: Remove unneeded validateDeadline method and update checkUserInput name

raynerlimfangyuh committed:
Deadline.java: Update findDeadlineDetails to catch ArrayIndexOutOfBoundsException too

raynerlimfangyuh committed:
Janet.java: Update code to use Parser.userInputChecker

weiliann commented on src/main/java/janet/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+        String[] deadlineDetails = findDeadlineDetails(commandDetails);     // (description, MMM dd yyyy hh:mm a)
+        String dateAndTimeString = deadlineDetails[1];    // MMM dd yyyy hh:mm a
+        DateTimeFormatter stringToDateTime = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mm a");
+        LocalDateTime dateAndTime = LocalDateTime.parse(dateAndTimeString, stringToDateTime);  // format String to LocalDateTime
```
This line is exceeding the line limit.

weiliann commented on src/main/java/janet/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+        super(description, symbol);
+        this.dueDate = dueDate;
+    }
+
```
Would be nice if there is only one blank line separating the code.

rayray39 replied:
Got it! Thanks

weiliann reviewed (commented):
Overall code is compliant with the Java coding standards 👍

raynerlimfangyuh committed:
build.gradle: Update dependencies to include javaFX

raynerlimfangyuh committed:
Add Main, DialogBox and Launcher for GUI

raynerlimfangyuh committed:
Ui.java: Update methods to return messages instead of print

raynerlimfangyuh committed:
Janet.java: Add runProgram method to return response

raynerlimfangyuh committed:
Ui.java: Remove print statements and HORIZONTAL_LINE

raynerlimfangyuh committed:
DialogBox.java: Update code to vertically center image in HBox

raynerlimfangyuh committed:
DialogBox.java: Update alignment of children to be CENTER_RIGHT

raynerlimfangyuh committed:
Janet.java: Remove run method for CLI and add startMessage method

raynerlimfangyuh committed:
Main.java: Add code to display janet welcome message

raynerlimfangyuh committed:
Janet.java: Add javadoc to startMessage method

hazzle23 commented on src/main/java/janet/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+
+    /**
+     * Returns a String array based on user's commands.
+     * First element = a String of janet.Deadline's description.
```
I like how your JavaDoc is very informative and well-structured!

rayray39 replied:
Thanks!

hazzle23 commented on src/main/java/janet/Janet.java:
```diff
@@ -0,0 +1,101 @@
+
+                CommandType commandType = Parser.getCommand(commandDetails);    // get the CommandType
+                switch (commandType) {
+                case BYE:
```
Good job on following the indentation convention for the `switch` statement!

hazzle23 commented on src/main/java/janet/Ui.java:
```diff
@@ -0,0 +1,167 @@
+    }
+
+
+    /**
+     * Prints the exit message.
+     */
+    public void exitMessage() {
```
I think you can consider naming this method with an action verbs instead, like `showExitMessage` or `displayExitMessage`.

rayray39 replied:
Good point! I must have missed that out

hazzle23 commented on src/test/java/janet/EventTest.java:
```diff
@@ -0,0 +1,37 @@
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Based on [this guide](https://se-education.org/guides/tutorials/intellijCodeStyle.html#tweak-import-order), I think static imports should be ordered first.

hazzle23 reviewed (commented):
Overall looks good to me!

raynerlimfangyuh committed:
Add MainWindow controller and view

raynerlimfangyuh committed:
Add DialogBox controller and view

raynerlimfangyuh committed:
Main.java: Refactor code to use javaFX FXML

raynerlimfangyuh committed:
DialogBox.java: Add javadoc to public methods

raynerlimfangyuh committed:
Merge branch 'Level-10'

raynerlimfangyuh committed:
Enable assertions

raynerlimfangyuh committed:
Parser.java: Add assert statement

Assume single task number to follow marking or delete commands

A marking or delete command works even when task numbers are separated by whitespaces

rayray39 committed:
Merge pull request #1 from rayray39/branch-A-Assertions

branch-A-Assertions

raynerlimfangyuh committed:
Parser.java: Abstract out complicated boolean expression

Boolean expression to check for invalid user command is too complicated to write in one line

Abstracting out the boolean expression into a public method makes it more readable

Let's abstract those complicated boolean expressions into their own public methods to make the code more readable

raynerlimfangyuh committed:
Parser.java: Abstract out complicated boolean expression

Boolean expressions to check for invalid or absent task number are too complicated and unreadable

Abstract out the boolean expression into public methods make them more readable

raynerlimfangyuh committed:
Refactor code by abstracting complicated boolean expressions

Too many boolean expressions to check for validity of user command

Many boolean expressions are being used in different areas of Parser.java to check for validity of user commands and this makes it hard to track and read.

Let's abstract the boolean expressions into methods to make them more readable and package it into a single method that can be easily called

rayray39 committed:
Merge pull request #2 from rayray39/branch-A-CodeQuality

branch-A-CodeQuality

raynerlimfangyuh committed:
Update method in TaskList.java to use varargs.

rayray39 committed:
Merge pull request #3 from rayray39/branch-A-Varargs

Update method in TaskList.java to use varargs.

raynerlimfangyuh committed:
CommandType.java: Add new CommandType value, VIEW

raynerlimfangyuh committed:
Add ScheduledTask class to track Tasks with date and time

raynerlimfangyuh committed:
Todo.java: Refactor code to use overloaded constructor

raynerlimfangyuh committed:
Task.java: Add scheduledDateTime attribute

raynerlimfangyuh committed:
Update scheduledDate attribute in Task to be LocalDate

raynerlimfangyuh committed:
Add view command

view command allows user to view tasks that are occuring on a particular date.

raynerlimfangyuh committed:
Add try catch block to parse dateAndTime

raynerlimfangyuh committed:
Update view command in TaskList.java to handle DateTimeParseException

raynerlimfangyuh committed:
Add exception handling for view command

Parser.java should handle all the exception handling and process invalid user commands. View command's exception handling is processed inside TaskList.java which is not the responsibility of TaskList.java.

Let's transfer the responsibility of processing the validity of view command from TaskList.java to Parser.java

raynerlimfangyuh committed:
Remove scheduledDate attribute from Task

ToDo class, which inherits Task, does not need access to the scheduledDate attribute. Only ScheduledTask inheritors, Deadline and Event, will require access to scheduledDate.

This removes the need to explicitly check for Deadline or Event objects in viewScheduledTasks method in TaskList. Makes the code more readable and maintanable.

Let's remove the scheduledDate attribute from Task and add it into ScheduledTask.

raynerlimfangyuh committed:
ToDo.java: Remove access to scheduledDate attribute

rayray39 committed:
Merge pull request #4 from rayray39/branch-B-ViewSchedules

branch-B-ViewSchedules

rayray39 committed:
Ui.java: Remove unneeded commented HORIZONTAL_LINE

rayray39 committed:
Ui.java: Remove unneeded commented HORIZONTAL_LINE

rayray39 committed:
Comment out testing code

rayray39 committed:
DeadlineTest.java: Update test methods

Update test methods to ensure that the invalid creation of Deadline objects will print the right error messages for the user to see.

rayray39 committed:
DeadlineTest.java: Update DueDate_ommision_test to correct expected output

rayray39 committed:
Deadline.java: Add comments for readability

rayray39 committed:
ParserTest.java: Update test methods

Update test methods to ensure that invalid user commands will return the correct error messages.

rayray39 committed:
Merge pull request #5 from rayray39/JUnit-Revision-1

JUnit-Revision-1

rayray39 committed:
Create gradle.yml for Continuous Integration (CI)

Let's add a gradle.yml file into ip/.github/workflows folder for continuous integration.

rayray39 committed:
Merge pull request #6 from rayray39/branch-A-CI

Create gradle.yml for Continuous Integration (CI)

rayray39 committed:
merge branch-C-Sort to master

rayray39 committed:
Add sort task command

Let's add sort task command for users to sort Deadline and Event objects according to their scheduledDate.

rayray39 committed:
Delete ScheduledTaskComparator.java

rayray39 committed:
Update sort command to sort by date and time

The sort command sorts Deadline and Event objects, respectively, based on their dates.

Let's update sort command to sort Deadline and Event objects by date and time in chronological order.

rayray39 committed:
Add getter for scheduledDate in LocalDate

rayray39 committed:
Ui.java: Add javadoc to methods

rayray39 committed:
Parser.java: Add checks for sort command

Let's add checks for sort command by creating methods that returns the results of boolean expressions so that the code is more readable and easy to trace.

rayray39 committed:
Parser.java: Resolve code problems identified

rayray39 committed:
Merge pull request #7 from rayray39/branch-C-Sort

branch-C-Sort

rayray39 committed:
Update creation of Tasks from .txt file

Storage.java handles the reading and creating of the Task objects, when loading from janet.txt file in directory. Storage.java also needs to check whether the content of the .txt file is valid.

Shifting the creation of the Task objects to the Task themselves ensure that the validity checks are also performed in the Task objects. This makes it easier to perform validity checks through exception handling.

rayray39 committed:
Update exception handling for loading of tasks

rayray39 committed:
Deadline.java: Add check when date is unexpected

Deadline object's dueDate can be earlier than today's date.

A check ensures that the dueDate specified by the user is not earlier than today's date.

Let's add a check in the creation of the Deadline object from user's input to ensure that the dueDate cannot be earlier than today's date.

rayray39 committed:
Event.java: Add check when date is unexpected

Event object's startDate can be earlier than today's date. The startDate can also be later than or equal to endDate

A check ensures that the startDate specified by the user is not earlier than today's date. The startDate is also not later than or equal to endDate.

Let's add a check in the creation of the Event object from user's input to ensure that the startDate cannot be earlier than today's date, and startDate cannot be later than or equal to endDate.

rayray39 committed:
Janet.java: Remove unneeded imports

rayray39 committed:
Storage.java: Remove unneeded imports

rayray39 committed:
TaskList.java: Remove unneeded imports

rayray39 committed:
Add javadoc to overloaded constructors

rayray39 committed:
Add javadoc to overloaded constructors

rayray39 committed:
ScheduledTask.java: Add javadoc to constructor

rayray39 committed:
Add javadoc to constructors

rayray39 committed:
TaskList.java: Add assert statement

rayray39 committed:
TaskList.java: Handle duplicated task description

TaskList.java allows tasks with the same descriptions to be added into the list of tasks.

Not allowing tasks with the same descriptions to be added makes it more user friendly because tasks should belong only to either a ToDo, Deadline or Event.

Let's add a check in the addTaskToList method in TaskList.java to prevent the addition of new tasks with descriptions that are identical to at least one other Task in the current list of tasks.

rayray39 committed:
DeadlineTest.java: Update actual print statement in test

rayray39 committed:
Parser.java: Update boolean variable to sound like boolean

rayray39 committed:
ParserTest.java: Add tests for sort and find commands

rayray39 committed:
Event.java: Update boolean expression

rayray39 committed:
EventTest.java: Add tests for invalid creation and valid creation from txt file

rayray39 committed:
DeadlineTest.java: Add tests for invalid due date and creation from text file

rayray39 committed:
Add setup in tests to ensure consistent time usage

rayray39 committed:
perform gradle clean

rayray39 committed:
Merge pull request #13 from rayray39/branch-A-MoreErrorHandling

A-MoreErrorHandling

rayray39 committed:
TaskList.java: Update javadoc of method

rayray39 committed:
Abstract out a method to find index of keyword in command

rayray39 committed:
ScheduledTask.java: Add javadoc to getIndexOfKeyword method

rayray39 committed:
Task.java: Update phrasing of header comments for getters

rayray39 committed:
Update boolean attribute in Task.java to sound more boolean

rayray39 committed:
Add title to stage in Main.java

rayray39 committed:
README.md: Add README content

rayray39 committed:
Ui.java: Update javadoc of method

rayray39 committed:
README.md: Add description for additional features

rayray39 committed:
README.md: Add quickstart section and overview of features

rayray39 committed:
Update README.md

rayray39 committed:
Merge pull request #14 from rayray39/branch-A-UserGuide

branch-A-UserGuide

rayray39 committed:
Add Ui.png

rayray39 committed:
README.md: Update README to embed janet.jar file link

rayray39 committed:
README.md: Update README

rayray39 committed:
Create JAR file

rayray39 committed:
Merge pull request #15 from rayray39/branch-A-Release

branch-A-Release

rayray39 committed:
README.md: Update README to embed latest Janet release

rayray39 committed:
TaskList.java: Add comment into method

rayray39 committed:
Update code to allow saving of tasks before today

rayray39 committed:
Update JUnit tests

rayray39 committed:
Storage.java: Update comment

rayray39 committed:
MainWindow.java: Update code to exit after bye command

rayray39 committed:
Deadline.java: Add condition to check for single due date/tima

rayray39 committed:
Deadline.java: Add check to prevent more than one due date and time

rayray39 committed:
Add check to ensure creation of event is valid
